### ISSUE TITLE
PHP 7.4 Trying to access array offset on value of type null

### ISSUE THREAD

**Issue by aun408** [HuasoFoundries/jpgraph#88]
**Affected File:** Amenadiel\JpGraph\Plot\PiePlot.php
**Line:** 398
**Reason:** The value of $this->csimtargets might null, accessing array offset on value of type null will cause exception (E_WARINING) in PHP 7.4

I would like to help and create a pull request but I am not sure what is the expecting value from $graph->legend->Add() if value of $this->csimtargets is null. 

My quick fix for now is check if $this->csimtargets is array, if it is not array then pass the null value to $graph->legend->Add() else using the original code ($this->csimtargets[$i]).

**Example**
```
if (!is_array($this->csimtargets)) {
    $csimtargetsTmp = null;
} else {
    $csimtargetsTmp = $this->csimtargets[$i];
}
$graph->legend->Add($l, $this->setslicecolors[$i % $numcolors], '', 0, $csimtargetsTmp, $alt, $wintarg);
```

**Comment by ffflabs** [comment]
Hmm you could use the coalescing operator

```
$csimtargetsTmp = ($this->csimtargets ?? [ ])[$i];
```

I'm not sure either what was supposed to go in that place. Most of this community edition is reverse engineering

**Comment by formikaio** [comment]
Created a [pull request](https://github.com/HuasoFoundries/jpgraph/pull/106) for it

**Comment by ffflabs** [comment]
Closed by #106
